parsers/TR.py

```diff
@@ -37,12 +39,28 @@
 }
 
 
+class LegacyHttpAdapter(adapters.HTTPAdapter):
+    def init_poolmanager(self, connections, maxsize, block=False):
+        ctx = ssl.create_default_context(ssl.Purpose.SERVER_AUTH)
+        ctx.options |= 0x4  # OP_LEGACY_SERVER_CONNECT
+        self.poolmanager = urllib3.poolmanager.PoolManager(
+            num_pools=connections, maxsize=maxsize, block=block, ssl_context=ctx
+        )
+
+
+# Use a LegacyHttpAdapter to avoid "unsafe legacy renegotiation disabled" error (issue #5484)
+# Original code source: https://stackoverflow.com/questions/71603314/ssl-error-unsafe-legacy-renegotiation-disabled
+def mount_legacy_adapter(session: Session):
+    session.mount("https://", LegacyHttpAdapter())
+
```

Comment:
@unitrium I noticed #5482 is suffering the same problem, do you think it would make sense to bring this into a lib instead?

Reply:
Yes, I think it would make sense to reuse this.